docs/selectors.md

```diff
@@ -84,9 +84,45 @@ Id engines are selecting based on the corresponding atrribute value. For example
 
 ### zs
 
-ZSelector is an experimental engine that tries to make selectors survive future refactorings. Example: `zs=div ~ "Login"`.
+Z-selector is an experimental engine to define selectors that can **survive future refactorings**. Changes in the UI markup can often cause existing UI selectors to break. While `data-*` attributes can be useful to establish contracts between app and test code, they can be an additional maintenance burden. It can also be difficult to introduce new attributes if you are testing third-party UI.
 
-TODO: write more.
+Z-selector aims to search elements with precision and resiliency against layout changes. It does not depend on strict nesting, and can localize element search through accessible selectors, like text content and type of element. It can then combine that with precise last-mile lookup.
```

Comment:
So is z selector a composite selector that supports both text and CSS selectors, that allows specifying element paths, without defining the actual relationship between nodes? (e.g. parent, sibling, “cousin”, grandparent, etc.) Can you use an xpath selector within a z selector?

Reply:
za is a superset of css, can't use xpath *inside* it. But composition `>>` works across all engines.

Reply:
Yeah, it’s just that the sample makes it seem like you can use the test selector engine “inside” of a z selector, so I was curious how to think about that. I like the “superset of CSS” thinking, which includes 1) the ability to select by text content, and the ability to define “flexible” node relationships.

Reply:
i'm hesitant to use the phrase "css superset", since we can't support ids (`#` is the ordinal selector), and there are some corner cases like #799. i've updated the content to answer the first question in this thread a little differently, at least for now.